### DISCUSSION TITLE
Stroybook v8 migration for MDX doc pages

### DISCUSSION THREAD

**Discussion by konsalex** [storybookjs/storybook#26043] [Help] [answered]
### Summary

Hey folks,

It is not super clear to me what should be the strategy for migrating MDX files for v8.

Currently we use MDX documentation pages and not for component doc.

An example is here: https://storybook-components-build.appspot.com/?path=/docs/getting-started-how-to-consume--docs

The code that is used there is plain MDX code, with imports of some components

```MDX
import { Meta, Unstyled } from '@storybook/addon-docs';
import { TabsState } from '@storybook/components';
import { Banner } from '@neo4j-ndl/react/src';

<Meta
  title="Getting started/How to consume"
  parameters={{
    viewMode: 'docs',
    previewTabs: {
      canvas: { hidden: true },
    },
  }}
/>

# How to use Needle 🪡

There are various ways to consume Needle 🪡 , and based on your use case you can choose the one that fits best your product's needs.

...
```

What is the way to migrate this page? When I start the server I get an error:

```
🚨 Unable to index ./src/StoryPages/Welcome.stories.mdx
```

and not sure how to proceed 🤔 

### Additional information

Storybook version: `8.0.0-beta.2`

The config (`main.ts`) is large, but the interesting parts:

```
 stories: [
    '../src/**/*.stories.mdx',
    '../src/**/*.stories.@(js|jsx|ts|tsx)',
  ],
  addons: [
    '@storybook/addon-links',
    '@storybook/addon-essentials',
    '@storybook/addon-a11y',
    '@storybook/addon-storysource',
  ],
```

**Comment by JReinhold** [comment]
1. I'm not sure if this is what's causing the error, but you need to change your imports from `@storybook/addon-docs` to `@storybook/blocks`. This is indirectly described here (not super clear though, I must admit) https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#addon-docs-removed-deprecated-blocksjs-entry
2. `Meta` doesn't support the `parameters` prop anymore, I'm not sure what it does for you in this case? See the API docs at https://storybook.js.org/docs/api/doc-block-meta

**Reply by konsalex** [reply, marked as answer]
@JReinhold found the issue. For some reason, maybe you know the internals, the `.stories.mdx` is not allowed anymore. If I have a name of `Welcome.mdx` or `Welcome.doc.mdx` it works as expected
